ids/id.go

```diff
@@ -32,11 +32,39 @@ var (
 // ID wraps a 32 byte hash used as an identifier
 type ID [IDLen]byte
 
+// Writeable is an helper which helps modifiying ID content
+func Writable(id *ID) []byte {
+	return id[:]
+}
+
+// Any modification to Bytes will be lost since id is passed-by-value
+// Use Writable(ID) is you need to modify the ID
+func (id ID) Bytes() []byte {
+	return id[:]
+}
+
```

Comment:
1. What's the reason for introducing these? It looks to me like we're trying to make it more verbose when you're modifying a copy vs a reference. I just wonder if these abstractions are really necessary or not. I feel like the existing pattern of using `id[:]` has been fine in my experience.
2. What is `Writable` for? I'm wondering in what world we're expecting to manually write over an `ID`.
3. It seems confusing that it is possible to get the bytes form of `id` using both `id[:]` and `id.Bytes()`. It would be nice if we could only make it possible to use one.

Reply:
I second Josh's questions -- I'm not sure what these changes buy us

Reply:
@danlaine I definitely should have kept this as a draft, as this is pretty exploratory for now.
I am trying to refactor ids.ID/NodeID/ShortID to remove duplication across them, and possibly change their internals.
Definiting these pendantc interfaces allows me to get more freedom in the refactoring. 
I am not sure I'll eventually keep the interfaces but they make the code change easier in the meantime